src/Nethermind/Nethermind.Merge.Plugin/Handlers/V1/ForkchoiceUpdatedV1Handler.cs

```diff
@@ -105,7 +105,7 @@ public ResultWrapper<ForkchoiceUpdatedV1Result> Handle(ForkchoiceStateV1 forkcho
             
             if (headUpdated && shouldUpdateHead)
             {
-                _poSSwitcher.ForkchoiceUpdated(newHeadBlock!.Header);
+                _poSSwitcher.ForkchoiceUpdated(newHeadBlock!.Header, finalizedHeader);
```

Comment:
finalizedHeader should be saved in FinalizationManager when we have FinalizationBlockHash != Keccak.Zero